LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs

```diff
@@ -144,11 +147,15 @@ public static NetworkServerConfiguration CreateFromEnvironmentVariables()
 
             // Create case insensitive dictionary from environment variables
             var envVars = new CaseInsensitiveEnvironmentVariables(Environment.GetEnvironmentVariables());
-
-            config.RunningAsIoTEdgeModule = !string.IsNullOrEmpty(envVars.GetEnvVar("IOTEDGE_APIVERSION", string.Empty));
+            config.ProcessingDelayInMilliseconds = envVars.GetEnvVar("PROCESSING_DELAY_IN_MS", Constants.DefaultProcessingDelayInMilliseconds);
+            config.RunningAsIoTEdgeModule = !envVars.GetEnvVar("CLOUD_DEPLOYMENT", false);
             config.IoTHubHostName = envVars.GetEnvVar("IOTEDGE_IOTHUBHOSTNAME", string.Empty);
             config.GatewayHostName = envVars.GetEnvVar("IOTEDGE_GATEWAYHOSTNAME", string.Empty);
-            config.EnableGateway = envVars.GetEnvVar("ENABLE_GATEWAY", config.EnableGateway);
+            config.EnableGateway = envVars.GetEnvVar("ENABLE_GATEWAY", config.EnableGateway) is bool enable_gateway
+                                   && !config.RunningAsIoTEdgeModule
+                                   && enable_gateway
+                                   ? throw new NotSupportedException("ENABLE_GATEWAY cannot be true if RunningAsIoTEdgeModule is false.")
+                                   : enable_gateway;
```

Comment:
[nit] Consider making the side effect here obvious - some parts of the logic are always true (e.g. `is bool enable_getway` is just used to assign a variable, but as `bool`s are value types it will always "match")

```suggestion
            config.EnableGateway = envVars.GetEnvVar("ENABLE_GATEWAY", true);
            if (!config.RunningAsIoTEdgeModule && config.EnableGateway)
            {
                throw new NotSupportedException("ENABLE_GATEWAY cannot be true if RunningAsIoTEdgeModule is false.");
            }
```